builds/codex-v103-aeturnum-monorepo/packages/neuro/src/codex_neuro/signal.py

```
@@ -0,0 +1,13 @@
+import numpy as np
+from scipy.signal import welch
+def psd(x, fs, nperseg=256):
+    f, Pxx = welch(np.asarray(x), fs=fs, nperseg=min(nperseg, len(x)))
+    return f.tolist(), Pxx.tolist()
+def bandpower(f, Pxx, lo, hi):
+    f = np.asarray(f); Pxx = np.asarray(Pxx)
+    idx = (f>=lo) & (f<=hi)
+    if not idx.any(): return 0.0
+    return float(np.trapz(Pxx[idx], f[idx]))
+def zscore(x):
+    x = np.asarray(x); mu=x.mean() if x.size else 0.0; sd=x.std() if x.size else 1.0
```

Ruff (0.14.3)

Multiple statements on one line (semicolon)

(E702)